tests/unit/add-one-test.js

```diff
@@ -25,3 +26,196 @@ test('add-one non-409 error', function (t) {
+  })
+})
+
+test('add-one validate rejects with Error (with message)', function (t) {
+  t.plan(2)
+
+  var state = {
+    validate: function () { return Promise.reject(new Error('Validation failed for the given docs')) }
+  }
+
+  var doc = {}
+  addOne(state, doc)
+
+  .then(function () {
+    t.fail('should throw an ValidationError')
+  })
+
+  .catch(function (error) {
+    t.is(error.name, 'ValidationError', 'validation error name matches')
+    t.is(error.message, 'Validation failed for the given docs', 'error message matches intent')
+  })
+})
+
+test('add-one validate rejects with a string', function (t) {
+  t.plan(2)
+
+  var state = {
+    validate: function () { return Promise.reject('Validation failed for the given docs') }
+  }
+
+  var doc = {}
+  addOne(state, doc)
+
+  .then(function () {
+    t.fail('should throw an ValidationError')
+  })
+
+  .catch(function (error) {
+    t.is(error.name, 'ValidationError', 'validation error name matches')
+    t.is(error.message, 'Validation failed for the given docs', 'error message matches intent')
+  })
+})
+
+test('add-one validate rejects with a value I', function (t) {
+  t.plan(3)
+
+  var state = {
+    validate: function () { return Promise.reject(false) }
+  }
+
+  var doc = {}
+  addOne(state, doc)
+
+  .then(function () {
+    t.fail('should throw an ValidationError')
+  })
+
+  .catch(function (error) {
+    t.is(error.name, 'ValidationError', 'validation error name matches')
+    t.is(error.message, 'check error value for more details', 'error message matches intent')
+    t.is(error.value, false, 'error.value is false')
+  })
+})
+
+test('add-one validate rejects with a value II', function (t) {
+  t.plan(3)
+
+  var state = {
+    validate: function () { return Promise.reject(1) }
+  }
+
+  var doc = {}
+  addOne(state, doc)
+
+  .then(function () {
+    t.fail('should throw an ValidationError')
+  })
+
+  .catch(function (error) {
+    t.is(error.name, 'ValidationError', 'validation error name matches')
+    t.is(error.message, 'check error value for more details', 'error message matches intent')
+    t.is(error.value, 1, 'error.value is 1')
+  })
+})
+
+test('add-one validate rejects with a value III', function (t) {
+  t.plan(4)
+
+  var state = {
+    validate: function () { return Promise.reject({ failure: true, tries: 1 }) }
+  }
+
+  var doc = {}
+  addOne(state, doc)
+
+  .then(function () {
+    t.fail('should throw an ValidationError')
+  })
+
+  .catch(function (error) {
+    t.is(error.name, 'ValidationError', 'validation error name matches')
+    t.is(error.message, 'check error value for more details', 'error message matches intent')
+    t.is(error.value.failure, true, 'error.value.failure is true')
+    t.is(error.value.tries, 1, 'error.value.tries is 1')
+  })
+})
+
+test('add-one validation fails with custom error', function (t) {
+  t.plan(4)
+
+  let customError = new Error('custom error message')
```

Comment:
we still do only use  `var` right now and don’t use any other es2016 features at this point

Reply:
that’s why the build is failing: https://travis-ci.org/hoodiehq/hoodie-store-client/jobs/273757512#L682

Reply:
Writing `let` has become reflexive for me 😅
Sorry about that. Can't we configure `standard` for style check regarding usage of ES6 variables?

Reply:
unfortunately not, that is one downside of `standard`. There were some discussions to do things like `standard-node4` etc but I don’t think it went anywhere. So for now, we have to catch these in our CIs and reviews :D

Reply:
Thank god you only have to look out for ES6 specifics. Reviews are tough. 
On that note, thanks for being patient with this PR, I'll fix this up today :)